ts/features/walletV3/payment/screens/WalletPaymentPickPspScreen.tsx

```diff
@@ -56,46 +89,128 @@ const WalletPaymentPickPspScreen = () => {
     navigation.navigate(WalletPaymentRoutes.WALLET_PAYMENT_MAIN, {
       screen: WalletPaymentRoutes.WALLET_PAYMENT_CONFIRM
     });
   };
 
+  const sortButtonProps: ListItemHeader["endElement"] = React.useMemo(
+    () => ({
+      type: "buttonLink",
+      componentProps: {
+        label: I18n.t("wallet.payment.psp.pspSortButton"),
+        accessibilityLabel: I18n.t("wallet.payment.psp.pspSortButton"),
+        onPress: present
+      }
+    }),
+    [present]
+  );
+
+  const pspSelected = pipe(selectedPspOption, O.toUndefined);
+
+  const SelectPspHeadingContent = React.useCallback(
+    () => (
+      <>
+        <H2>{I18n.t("wallet.payment.psp.title")}</H2>
+        <VSpacer size={16} />
+        <Body>{I18n.t("wallet.payment.psp.description")}</Body>
+        <Body>
+          {I18n.t("wallet.payment.psp.taxDescription")}{" "}
+          <Body weight="SemiBold">
+            {I18n.t("wallet.payment.psp.taxDescriptionBold")}
+          </Body>
+        </Body>
+        <VSpacer size={16} />
+        <ListItemHeader
+          label={I18n.t("wallet.payment.psp.pspTitle")}
+          accessibilityLabel={I18n.t("wallet.payment.psp.pspTitle")}
+          endElement={sortButtonProps}
+        />
+      </>
+    ),
+    [sortButtonProps]
+  );
+
   return (
-    <BaseScreenComponent goBack={true}>
-      <GradientScrollView
-        primaryActionProps={{
-          label: "Continua",
-          accessibilityLabel: "Continua",
-          onPress: handleContinue,
-          disabled: isLoading,
-          loading: isLoading
-        }}
-      >
-        <DebugPrettyPrint title="pspListPot" data={pspListPot} />
-      </GradientScrollView>
-    </BaseScreenComponent>
+    <GradientScrollView
+      primaryActionProps={{
+        label: I18n.t("wallet.payment.psp.continueButton"),
+        accessibilityLabel: I18n.t("wallet.payment.psp.continueButton"),
+        onPress: handleContinue,
+        disabled: isLoading || !canContinue,
+        loading: isLoading
+      }}
+    >
+      <SelectPspHeadingContent />
+      {!isLoading && (
+        <RadioGroup<string>
+          onPress={handlePspSelection}
+          type="radioListItemWithAmount"
+          selectedItem={pspSelected?.idBundle}
+          items={getRadioItemsFromPspList(sortedPspList, showFeaturedPsp)}
+        />
+      )}
+      {isLoading && <WalletPspListSkeleton />}
+      {sortPspBottomSheet}
+    </GradientScrollView>
   );
 };
 
+const getRadioItemsFromPspList = (
+  pspList?: Array<Bundle>,
+  showFeaturedPsp?: boolean
+) =>
+  !pspList
+    ? []
+    : pspList?.map((psp, index) => ({
+        id: psp.idBundle ?? index.toString(),
+        label: psp.bundleName ?? I18n.t("wallet.payment.psp.defaultName"),
+        isSuggested: psp.onUs && showFeaturedPsp,
+        suggestReason: I18n.t("wallet.payment.psp.featuredReason"),
+        formattedAmountString: formatNumberCentsToAmount(
+          psp.taxPayerFee || 0,
+          true,
+          "right"
+        )
+      }));
+
```

Comment:
This could be simplified using `fp-ts`, what do you think??

Reply:
Addressed into 8f9bd167bab8c4e8883c76eee21bdfd339c9ddc6, let me know what do you think about it 👍